### ISSUE TITLE
max range of interior lights and possibly spotlights

### ISSUE THREAD

**Issue by Krienas** [KeenSoftwareHouse/SpaceEngineers#336]
Hello guys. I really want to increase max range limitation for interior lights and possibly spotlights also. Reason - properly functioning lights are almost critical in DX11 with sun rotation . With 10m range larger areas (for example hangars) remain dark because 10m range is too short. It wasn't a problem with DX9 as whole game was lighter - light was more aesthetics than functional necessity. This change could also keep us within DX11 and give time for those who work on proper lighting solution. 

Atm am thinking to increase max range to 20m. If there are reasons for different max range values, would like to hear them.

**Comment by joemorin73** [comment]
I believe you can actually set this in a Mod.  I believe I recall seeing a mod bring the max range up to 15m.

**Comment by joemorin73** [comment]
This mod adds light that are of a larger range:
http://steamcommunity.com/sharedfiles/filedetails/?id=441277694

This mod will increase the stock mods:
http://steamcommunity.com/sharedfiles/filedetails/?id=379581116

**Comment by Krienas** [comment]
I do not think we should need mods for this. I see it as a bug which became problem in DX11 with lighting/shadows done more space like and sun rotation added (station is no longer permanently lighted), but lights themselves not adapted. Currently lights are OK for narrow spaces of 1-2 large blocks. For 3 blocks can be useful in some visual cases, but not really functional as light. And if the light is 4 blocks above on ceiling, objects on floor are black. You know, when I saw it for the first time I thought it is misunderstanding and lights are not working at all. That was totally counter-intuitive :)

Of course if there were no possibility to make it part of the game we could create/use mod to fix it. That is what we used to do in the past, when source was not available. Alternatively we could try to double amount of lights everywhere (if we want to stick to vanilla)... But admit, that both approaches are contributing to lag workarounds, not solutions. And they are workarounds for problem which should not exit in vanilla in the first place. Also, I believe, that amount of mods you may have in game is limited. So there are better ways to use them (for add-ons, enriching experience, not making it possible). :)

**Comment by Tyrsis** [comment]
I think this sort of lighting issue is actually a bug, and should be reported in forums.  I also believe it's actually already been reported.

**Comment by devu** [comment]
Sure that lights in DX11 are awful. But be careful what you wish for when
asking for spotlights. The way it works is simply not performance friendly
solution. Increasing a range in vanilla simply would result in having extra
mod to decrease it to be able to play on DS.
Keen already have made mistake convincing all single players that there is
no limit to what computer can handle.

**Comment by Krienas** [comment]
I do not think we should change default values, so basically when you drag light into the world it will work as it used to work before. Am talking about max range limitation, so there was possibility to increase range where it is needed.

**Comment by joemorin73** [comment]
If I recall correctly, the spotlight performance issue is only in DX9 and not in DX11.  Would it now be advantageous to try to improve DX9 spotlight?
